### DISCUSSION TITLE
Puzzles (simon g tatham's portable puzzle collection) is not creating a Launchpad Icon -- should it?

### DISCUSSION THREAD

**Discussion by thegushi** [Homebrew/discussions#3885] [Casks]
Hey there all,

Before I open a bug report, I wanted to ask: is this the expected behavior? -- I don't know how to read the Brew files so I don't know if a brew-installed app is supposed to show up in launchpad or not.  Is there some obvious way to tell?

If it should be doing this, and it's not, please let me know and I'll create the appropriate bug.  I'm familiar with FreeBSD's ports and pkgs, but I'm just a user as far as brew goes.

Take care!

**Comment by SMillerDev** [comment]
That completely depends on how upstream configured their install and what you're installing exactly. I would assume that in FreeBSD OpenSSL does not create an icon but Google Chrome would.

**Reply by thegushi** [reply]
Sorry, let me be more clear.  I'm on MacOS.  I mentioned FreeBSD because on that system, the ports/pkg system is all Makefiles, so there are analogs I can understand, but it's totally not the same.

The cask downloads a single dmg, with a single .app in it, which is a Gui app which has macOS-style menus.  I can evoke it from a command prompt and it works.  If I had downloaded the .dmg manually, that's what I would have done, dragged it to Applications.

It's not a complicated cask: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/puzzles.rb

(For what it's worth, it has not created a Launchpad icon on either of my macs, both running Ventura, both of which pass brew doctor cleanly)

**Reply by SMillerDev** [reply]
I think the only way to know would be to download the dmg and see if it shows up in launchpad. Casks are generally just moving some files or running an upstream installer so it's rarely the case that it's missing a step, but it's impossible to know for sure unless you try.

**Reply by thegushi** [reply]
Aah, it seems I was installing the formula, not the cask.  The cask creates a desktop icon, the formula does not.  I don't know if that, itself, is a bug, but my issue's resolved.  Thanks!
